### PR TITLE
110/account destroy

### PR DESCRIPTION
**Pull request by thomasjinlo**
In `destroy.js` it checks whether or not the user is signed in. Depending on that it'll make the request or just return a resolved promise after clearing session. If this can be better refactored please let me know.

### CODE REVIEW

File: lib/destroy.js [hunk 1]
```diff
@@ -28,6 +42,6 @@ function destroy (state) {
 
     delete state.account
 
-    return clone(accountProperties)
-  })
+    return Promise.resolve(clone(accountProperties))
+  }
 }
```
Comment (gr2m) [inline lib/destroy.js]: You can simplify this code by creating a promise variable and then it with the request only if the user is signed in. Like so

```js
var promise = Promise.resolve()

if (internals.isSignedIn(state)) {
  promise = promise.then(function () {
    internals.request({
      method: 'DELETE',
      url: state.url + '/session/account',
      headers: {
        authorization: 'Session ' + state.account.session.id
      }
    })
  })
}

return promise.then(function () {
  internals.clearSession({
    cacheKey: state.cacheKey
  })

  state.emitter.emit('signout', clone(state.account))
  state.emitter.emit('destroy', clone(state.account))

  delete state.account

  return clone(accountProperties)
})
```

Does that make sense?

File: test/integration/destroy-test.js [hunk 2]
```diff
@@ -60,3 +61,46 @@ test('destroy account', function (t) {
...
+  // mock the Date object to always return 1970-01-01T00:00:00.000Z
+  var clock = lolex.install(0)
+  var account = new Account({
+    url: baseURL,
+    id: 'abc4567'
+  })
+
+  nock(baseURL)
+  .delete('/session/account')
+  .reply(204)
```
Comment (gr2m) [inline test/integration/destroy-test.js]: you can remove these lines, we don’t need to mock anything, because we don’t send any request to the server

Comment (thomasjinlo) [inline test/integration/destroy-test.js]: Oh woopsie, you are totally right. I'll fix it in a jiffy.

File: lib/destroy.js (hunk 1 above)
Comment (thomasjinlo) [inline lib/destroy.js]: Yes I think I understand, it's pretty interesting I have not seen this before. It looks like if the user is signed in you are chaining two `.then()`. I'm assuming it's possible to chain them.

Comment (gr2m) [inline lib/destroy.js]: I can highly recommend this blog post for lots of interesting gotchas about Promises:
https://pouchdb.com/2015/05/18/we-have-a-problem-with-promises.html

**Comment by gr2m** [comment]
released as [v4.4.0](https://github.com/hoodiehq/hoodie-account-client/releases/tag/v4.4.0)

**Pull request merged**